src/mpl/src/gpu/mpl_gpu_cuda.c

```diff
@@ -327,15 +327,17 @@ int MPL_gpu_free_hook_register(void (*free_hook) (void *dptr))
 CUresult CUDAAPI cuMemFree(CUdeviceptr dptr)
 {
     CUresult result;
-    gpu_free_hooks_cb((void *) dptr);
+    if ((void *) dptr != NULL)
```

Comment:
make the condition consistent. There is no need to do cast here. Also, I would prefer to always use explicit braces.